bynary/apps/demo/src/app/core/feature/recipe/pages/detail/recipe-detail.component.spec.ts

```diff
@@ -138,43 +140,80 @@ describe('RecipeDetailComponent', () => {
                 expect(recipeDescription.nativeElement.textContent).toEqual(' Bla bla ');
             });
 
-            it('should have a ingredients', () => {
+            it('should have ingredients', () => {
                 const recipeIngredients = debugElement.queryAll(By.css('.c-recipe-detail-component__recipe-ingredients'));
 
                 expect(recipeIngredients).toHaveLength(2);
-                expect(recipeIngredients[0].nativeElement.textContent).toEqual(' Schnitzel - 2 ');
-                expect(recipeIngredients[1].nativeElement.textContent).toEqual(' Pommes - 1 ');
             });
 
-        });
+            describe('ingredients', () => {
+                let recipeIngredients: DebugElement[];
 
-        // TODO: Überprüfen, ob Buttons richtige Texte anzeigen
+                beforeEach(() => {
+                    recipeIngredients = debugElement.queryAll(By.css('.c-recipe-detail-component__recipe-ingredients'));
+                });
 
-        it('should call onAddToShoppingList method when button "To Shopping List" is clicked', () => {
-            const addToShoppingList = jest.spyOn(component, 'onAddToShoppingList');
-            const toShoppingListButton = debugElement.query(By.css('#toShoppingList'));
+                it('should contain ingredient Schnitzel (2)', function () {
+                    expect(recipeIngredients[0].nativeElement.textContent).toEqual(' Schnitzel - 2 ');
+                });
 
-            toShoppingListButton.triggerEventHandler('click', addToShoppingList);
+                it('should contain ingredient Pommes (1)', function () {
+                    expect(recipeIngredients[1].nativeElement.textContent).toEqual(' Pommes - 1 ');
+                });
+
+            });
 
-            expect(addToShoppingList).toHaveBeenCalledTimes(1);
         });
 
-        it('should call onEditRecipe method when button "Edit Recipe" is clicked', () => {
-            const onEditRecipe = jest.spyOn(component, 'onEditRecipe');
-            const editRecipeButton = debugElement.query(By.css('#editRecipe'));
+        describe('buttons', () => {
+            let deleteRecipeBtn: DebugElement;
+            let toShoppingListBtn: DebugElement;
+            let editRecipeBtn: DebugElement;
 
-            editRecipeButton.triggerEventHandler('click', onEditRecipe);
+            beforeEach(() => {
+                toShoppingListBtn = debugElement.query(By.css('#toShoppingList'));
+                editRecipeBtn = debugElement.query(By.css('#editRecipe'));
+                deleteRecipeBtn = debugElement.query(By.css('#deleteRecipe'));
+            });
 
-            expect(onEditRecipe).toHaveBeenCalledTimes(1);
-        });
+            it('should call onAddToShoppingList method when button "To Shopping List" is clicked', () => {
+                const addToShoppingList = jest.spyOn(component, 'onAddToShoppingList');
+
+                toShoppingListBtn.triggerEventHandler('click', addToShoppingList);
```

Comment:
Das könntest du so machen, nur müsstest du dann nicht auch den Spy übergeben, also in diesem Fall `addToShoppingList`. 
Der `triggerEventHandler` nimmt als zweites Argument ein `eventObjekt` wie z.B. ein MouseEvent oder andere Parameter welche als `clickEvent` übergeben werden sollen. Hier übergibst du `addToShoppingList`, du könntest es aber auch mit einem leeren Objekt austauschen, da du nur den 'click' selbst bräuchtest.

Besser wäre nur direkt den 'click' auszuführen ohne noch Objekte mit zu übergeben.
Dazu bräuchtest du wieder das `nativeElement`.  
Da du dieses auch nutzt wenn du Text des Buttons überprüfst, kannst du das auch gleich am Anfang des Blocks deklarieren.

z.B. `let toShoppingListBtn: HTMLElemnt;`

in der beforeEach() dann zuweisen: 

`toShoppingListBtn = debugElemnt.query(By.css('#toShoppingList')).nativeElement as HTMLElement;`
( Hier ist das casting `as HTMLElement` nötig, da `.nativeElement` nur `any` zurück gibt)


und den Click dann ausführen im Test:
`toShoppingListBtn.click()`